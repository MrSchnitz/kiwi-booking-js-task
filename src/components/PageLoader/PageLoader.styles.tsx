import styled from "styled-components";

export const PageLoaderWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  min-height: 100vh;
`;
