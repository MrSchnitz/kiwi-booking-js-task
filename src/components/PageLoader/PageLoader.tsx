import React from "react";
import { PageLoaderWrapper } from "./PageLoader.styles";
import { Loading } from "@kiwicom/orbit-components";

const PageLoader = React.memo(() => {
  return (
    <PageLoaderWrapper>
      <Loading type="pageLoader" />
    </PageLoaderWrapper>
  );
});

export default PageLoader;
