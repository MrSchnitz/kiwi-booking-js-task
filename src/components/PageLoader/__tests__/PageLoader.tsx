import { render } from "@testing-library/react";
import PageLoader from "../PageLoader";

describe("PageLoader", () => {
  it("should render basic component", () => {
    const { container } = render(<PageLoader />);

    expect(container).toMatchSnapshot();
  });
});
