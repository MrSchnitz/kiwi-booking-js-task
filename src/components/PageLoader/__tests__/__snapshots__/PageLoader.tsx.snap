// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader should render basic component 1`] = `
<div>
  <div
    class="sc-bczRLJ hfDjeB"
  >
    <div
      class="Loading__StyledLoading-sc-199xidk-0 bMyNJA"
    >
      <svg
        class="Loading__StyledSpinner-sc-199xidk-2 fBxWoj"
        viewBox="0 0 40 40"
      >
        <circle
          class="Loading__StyledSpinnerCircle-sc-199xidk-3 emBrZC"
          cx="50%"
          cy="50%"
          r="18"
          type="pageLoader"
        />
      </svg>
      <div
        class="Loading__StyledLoadingText-sc-199xidk-1 cdYraH"
        type="pageLoader"
      />
    </div>
  </div>
</div>
`;
